src/VbsEnclaveSDK/src/veil_enclave_lib/userboundkey.vtl1.cpp

```diff
@@ -0,0 +1,589 @@
+        _In_ const veil::vtl1::developer_types::keyCredentialCacheConfig& cache_config)
+    {
+        // Convert cache_config to the correct type
+        auto abi_cache_config = veil::vtl1::userboundkey::ConvertCacheConfig(cache_config);
+
+        return veil_abi::Untrusted::Stubs::userboundkey_establish_session_for_create(
+            reinterpret_cast<uintptr_t>(enclave),
+            key_name,
+            ecdh_protocol,
+            message,
+            window_id,
+            abi_cache_config);
+    }
+
+    veil_abi::Types::credentialAndFormattedKeyNameAndSessionInfo userboundkey_establish_session_for_load(
+        _In_ const void* enclave,
+        _In_ const std::wstring& key_name,
+        _In_ const std::wstring& message, 
+        _In_ const uintptr_t window_id,
+        _In_ const uint64_t nonce)
+    {
+        return veil_abi::Untrusted::Stubs::userboundkey_establish_session_for_load(
+            reinterpret_cast<uintptr_t>(enclave),
+            key_name,
+            message,
+            window_id,
+            nonce);
+    }
+
+    // New function to extract secret and authorization context from credential
+    std::vector<std::uint8_t> userboundkey_get_authorization_context_from_credential(
+        _In_ const std::vector<std::uint8_t>& credential_vector,
+        _In_ const std::vector<std::uint8_t>& public_key,
+        _In_ const std::wstring& message,
+        _In_ const uintptr_t window_id)
+    {
+        return veil_abi::Untrusted::Stubs::userboundkey_get_authorization_context_from_credential(
+            credential_vector,
+            public_key,
+            message,
+            window_id);
+    }
+
+    std::vector<std::uint8_t> userboundkey_get_secret_from_credential(
+        _In_ const std::vector<std::uint8_t>& credential_vector,
+        _In_ const std::vector<std::uint8_t>& public_key,
+        _In_ const std::wstring& message,
+        _In_ const uintptr_t window_id)
+    {
+        return veil_abi::Untrusted::Stubs::userboundkey_get_secret_from_credential(
+            credential_vector,
+            public_key,
+            message,
+            window_id);
+    }
+}
+
+namespace veil::vtl1::userboundkey
+{
+// RAII wrapper using WIL for USER_BOUND_KEY_AUTH_CONTEXT_HANDLE to prevent resource leaks
+namespace
+{
+    inline void close_auth_context_handle(USER_BOUND_KEY_AUTH_CONTEXT_HANDLE handle) noexcept
+    {
+        if (handle)
+        {
+            // CloseUserBoundKeyAuthContext returns HRESULT, but we're in a noexcept context
+            // so we ignore the return value (similar to the original implementation)
+            (void)CloseUserBoundKeyAuthContext(handle);
+        }
+    }
+}
+
+using unique_auth_context_handle = wil::unique_any<
+    USER_BOUND_KEY_AUTH_CONTEXT_HANDLE,
+    decltype(&close_auth_context_handle),
+    close_auth_context_handle,
+    wil::details::pointer_access_all,
+    USER_BOUND_KEY_AUTH_CONTEXT_HANDLE,
+    decltype(nullptr),
+    nullptr
+>;
```

Comment:
Why so elaborate?  Can this just be one-liner like this,

`using unique_sessionhandle = wil::unique_any<USER_BOUND_KEY_SESSION_HANDLE, decltype(&::CloseUserBoundKeySession), ::CloseUserBoundKeySession>;`